src/OrchardCore/OrchardCore.Media.Core/OrchardCore.Media.Core/DefaultMediaFileStore.cs

```diff
@@ -77,7 +92,15 @@ public Task<Stream> GetFileStreamAsync(IFileStoreEntry fileStoreEntry)
 
         public Task CreateFileFromStreamAsync(string path, Stream inputStream, bool overwrite = false)
         {
-            return _fileStore.CreateFileFromStreamAsync(path, inputStream, overwrite);
+            MediaCreationContext mediaContext = new MediaCreationContext
+            {
+                Path = path,
+                Stream = inputStream
+            };
+            _mediaEventHandlers.Invoke((handler, context) => handler.MediaCreating(context), mediaContext, Logger);
+            var task = _fileStore.CreateFileFromStreamAsync(path, mediaContext.Stream, overwrite);
+            _mediaEventHandlers.Invoke((handler, context) => handler.MediaCreated(context), mediaContext, Logger);
+            return task;
```

Comment:
async. plus the `_fileStore.CreateFileFromStreamAsync` should read the path from the mediaContext. As an event handler might desire to manipulate this.